patches/server/0988-fix-MC-65198-ITEM_CRAFTED-statistic-isn-t-always-inc.patch

```diff
@@ -0,0 +1,56 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: elmital <54907162+elmital@users.noreply.github.com>
+Date: Fri, 14 Jul 2023 18:17:38 +0200
+Subject: [PATCH] fix: MC-65198 ITEM_CRAFTED statistic isn't always incremented
+ correctly
+
+
+diff --git a/src/main/java/net/minecraft/world/inventory/ItemCombinerMenu.java b/src/main/java/net/minecraft/world/inventory/ItemCombinerMenu.java
+index ff770b9ce68a62418de0c7ed389650626fa1dcb2..e939bbc42216d688be655c47fff362906ed5f3f4 100644
+--- a/src/main/java/net/minecraft/world/inventory/ItemCombinerMenu.java
++++ b/src/main/java/net/minecraft/world/inventory/ItemCombinerMenu.java
+@@ -73,6 +73,13 @@ public abstract class ItemCombinerMenu extends AbstractContainerMenu {
+             public void onTake(Player player, ItemStack stack) {
+                 ItemCombinerMenu.this.onTake(player, stack);
+             }
++
++            // Paper start - fix MC-65198 shift clicking in result slot of a smiting table
++            @Override
++            protected void onQuickCraft(ItemStack stack, int amount) {
++                stack.onCraftedBy(player.level(), player, amount);
++            }
++            // Paper end
+         });
+     }
+ 
+diff --git a/src/main/java/net/minecraft/world/inventory/ResultSlot.java b/src/main/java/net/minecraft/world/inventory/ResultSlot.java
+index ddf87ef1979c4ae4f14cd6aa220f9146d5e20909..4a8529ebb1bfce015bd77a35076fcc3bee33fa90 100644
+--- a/src/main/java/net/minecraft/world/inventory/ResultSlot.java
++++ b/src/main/java/net/minecraft/world/inventory/ResultSlot.java
+@@ -45,7 +45,7 @@ public class ResultSlot extends Slot {
+     @Override
+     protected void checkTakeAchievements(ItemStack stack) {
+         if (this.removeCount > 0) {
+-            stack.onCraftedBy(this.player.level(), this.player, this.removeCount);
++            stack.onCraftedBy(this.player.level(), this.player, stack.getCount()); // Paper - fix MC-65198 for drop action in result slot
+         }
+ 
+         Container var3 = this.container;
+diff --git a/src/main/java/net/minecraft/world/inventory/StonecutterMenu.java b/src/main/java/net/minecraft/world/inventory/StonecutterMenu.java
+index 1b1f814770d1a906ed880df578845be2e9a14f46..ae1b53e1e7185a6af326c274255803fc31b6ca36 100644
+--- a/src/main/java/net/minecraft/world/inventory/StonecutterMenu.java
++++ b/src/main/java/net/minecraft/world/inventory/StonecutterMenu.java
+@@ -114,6 +114,13 @@ public class StonecutterMenu extends AbstractContainerMenu {
+                 super.onTake(player, stack);
+             }
+ 
++            // Paper start - fix MC-65198 shift clicking in the result slot of StoneCutter
++            @Override
++            protected void onQuickCraft(ItemStack stack, int amount) {
++                stack.onCraftedBy(level, playerInventory.player, amount);
++            }
++            // Paper end
```

Comment:
I think this is missing some of the other logic, like awarding recipes perhaps?

Reply:
Like I explain above similar methods are called twice so here the recipes awarding is already called later in the StoneCutter#quickMoveStack(Player, int) method.

Reply:
Yep, I see that now.